### PR TITLE
Add support for using bazelisk --migrate to test incompatible flags

### PR DESCRIPTION
**Pull request by meteorcloudy**
@laurentlb

### CODE REVIEW

File: buildkite/aggregate_incompatible_flags_test_result.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+   def __init__(self, job):
+      threading.Thread.__init__(self)
+      self.job = job
+      self.log = None
+
+   def run(self):
+      self.log = get_build_log(self.job)
+
+
+def buildkite_token():
```
Comment (fweikert) [inline buildkite/aggregate_incompatible_flags_test_result.py]: You could refactor bazelci.saucelabs_token() to accept key and input, and then just call it here.

File: buildkite/aggregate_incompatible_flags_test_result.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+
+def get_build_log(job):
+    log_url = job["raw_log_url"]
+    output = urllib.request.urlopen(get_build_log_api_url(log_url)).read().decode("utf-8")
+    return output
+
+
+def process_build_log(failed_jobs_per_flag, log, job):
+    if log.find("Success: No migration needed.") != -1:
```
Comment (fweikert) [inline buildkite/aggregate_incompatible_flags_test_result.py]: why not just 
if "Success: No migration needed." in log:

(and below).

File: buildkite/aggregate_incompatible_flags_test_result.py (hunk 1 above)
Comment (meteorcloudy) [reply buildkite/aggregate_incompatible_flags_test_result.py]: I believe the SAUCELABS_TOKEN is different from BUILDKITE_API_TOKEN?

File: buildkite/aggregate_incompatible_flags_test_result.py (hunk 2 above)
Comment (meteorcloudy) [reply buildkite/aggregate_incompatible_flags_test_result.py]: Done, thanks!

**Comment by meteorcloudy** [comment]
@fweikert @philwo 
Looks like the build kite token we use on CI doesn't have read access to job log?
https://buildkite.com/bazel/bazelisk-plus-incompatible-flags/builds/14#8dacee4b-3134-4f25-8373-f654b21ef069

Can your grant the access to the token>?

**Comment by philwo** [comment]
@meteorcloudy Yes, I'll grant access shortly. Looking into it right now.

**Comment by philwo** [comment]
@meteorcloudy Done!

**Comment by meteorcloudy** [comment]
@philwo Works like a charm!

**Pull request merged**